### PR TITLE
Taylor-31-SignUpForm-FirebaseConnect

### PR DESCRIPTION
**Pull request by TaylorDJones11**
## What type of PR is this? (check all applicable)

- [ ] Feature
- [ ] Refactor
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Other: Connecting Form to Firebase 
  
  
## Description of what this Pull Request does
- Includes the Signup Page which displays the Signup Form  and Customizable Aside
- The Signup form will collect the name, email, and password of the manager.
- The Signup form will store the manager's password and email in the Firebase database. 
- The PR included adding error messages in the form along with minor styling changes, including removing Customizable Avatar and changing the size of the form. 
- 

## Related Tickets & Documents
- Closes  https://github.com/AritDeveloperCircle/findyourpeople/issues/87
- Related Issues: (enter ticket numbers here)
  
## QA Instructions, Screenshots, Recordings
Screenshot 2023-05-14 at 18 39 57
Screenshot 2023-05-15 at 13 55 50

  
## Added/updated tests?
- [ ] Yes
- [ ] Not Yet, and here is why: __explain here__
- [ ] I need help with tests
- [x] No, and here is why: Testing won't be done

### CODE REVIEW

File: src/components/signupform/SignupForm.js [hunk 1]
```diff
@@ -1,110 +1,145 @@
...
 
+  // Validate Submit Form
   const validateForm = () => {
-    if (!user.name || !user.email || !user.password) {
-      alert('please fill out all required fields.')
-      return false
+    const errors = {};
+
+    // Validate Name
+    const namePattern = /^[a-zA-Z]{2,50}(?:\s+[a-zA-Z]{2,50})+$/;
```
Comment (jjleigh) [inline src/components/signupform/SignupForm.js:41]: Create a validation module in a helper folder and move these validation-related functions there. Then import the functions from that module here and just call them with the necessary parameters.

File: src/composables/authSignUp.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        updateProfile(userCredential.user, { displayName: name });
+      })
+      .then((userCredential) => {
+        const user = userCredential.user;
+        setAddUser(user);
+        console.log("user signed up", user);
+      })
+      .catch((error) => {
+        const errorCode = error.code;
+        const errorMessage = error.message;
```
Comment (jjleigh) [inline src/composables/authSignUp.js:20]: Did you want to console log the error code and message here?

File: src/pages/signup.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import CustomizableAside from "@/components/common/customizableAside";
+import SignupFormContainer from "@/components/signupform/SignupFormContainer";
+
+export default function signup() {
```
Comment (jjleigh) [inline src/pages/signup.js:4]: I love how this file looks. Short, sweet and easy to read. Very well done!  🥳

**Review by jjleigh — CHANGES_REQUESTED** [review]

File: src/components/signupform/SignupForm.js (hunk 1 above)
Comment (TaylorDJones11) [reply src/components/signupform/SignupForm.js]: I've updated the file @jjleigh

File: src/composables/authSignUp.js (hunk 2 above)
Comment (jjleigh) [reply src/composables/authSignUp.js:19]: @TaylorDJones11 You are just assigning variables here but not using them.

Comment (TaylorDJones11) [reply src/composables/authSignUp.js]: I've updated this as well @jjleigh

**Review by jjleigh — APPROVED** [review]

**Comment by TaylorDJones11** [comment]
@msarit I need your review to continue on with the styling of the form

File: src/components/signupform/helperFunctions/regexValidation.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  if (!user.password) {
+    errors.password = "Please enter a password";
+  } else if (user.password.length < 8 || user.password.length > 20) {
+    errors.password = "Please enter a password between 8 and 20 characters";
+  } else if (!user.password.match(passwordPattern)) {
+    errors.password =
+      "Password must contain one digit, one lowercase letter, one uppercase letter, and one special character";
+  }
+
+  // Check if user exists
```
Comment (msarit) [inline src/components/signupform/helperFunctions/regexValidation.js:34]: @TaylorDJones11 Why is this code commented?

**Review by msarit — APPROVED** [review]

**Pull request merged**